packages/djvulibre.rb

```diff
@@ -0,0 +1,50 @@
+# Adapted from Arch Linux djvulibre PKGBUILD at:
+# https://github.com/archlinux/svntogit-packages/raw/packages/djvulibre/trunk/PKGBUILD
+
+require 'package'
+
+class Djvulibre < Package
+  description 'Suite to create, manipulate and view DjVu déjà vu documents'
+  homepage 'http://djvu.sourceforge.net/'
+  version '3.5.28'
+  license 'GPL2'
+  compatibility 'all'
+  source_url "https://downloads.sourceforge.net/project/djvu/DjVuLibre/#{version}/djvulibre-#{version}.tar.gz"
+  source_sha256 'fcd009ea7654fde5a83600eb80757bd3a76998e47d13c66b54c8db849f8f2edc'
+
+  binary_url({
+    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/djvulibre-3.5.28-chromeos-armv7l.tar.xz',
+     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/djvulibre-3.5.28-chromeos-armv7l.tar.xz',
+       i686: 'https://dl.bintray.com/chromebrew/chromebrew/djvulibre-3.5.28-chromeos-i686.tar.xz',
+     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/djvulibre-3.5.28-chromeos-x86_64.tar.xz'
+  })
+  binary_sha256({
+    aarch64: '2a662b5b95e3e5c24531cb4ae0f5b33c044895f33fc88c562026a4124ff15295',
+     armv7l: '2a662b5b95e3e5c24531cb4ae0f5b33c044895f33fc88c562026a4124ff15295',
+       i686: 'a0cf591f8e04fc89a0c2f45f0ad67c7c907971dcf85371a2184c3ea54e605da4',
+     x86_64: 'd17ec121e01aea869ea74b421aebcf4b8f019e8b32268859256aadd32019aab5'
+  })
+ 
+  depends_on 'libjpeg'
+  depends_on 'libtiff'
```

Comment:
Doesn't libtiff include libjpeg as a dependency?  This means libjpeg is a child dependency and doesn't need to be listed, right?

Reply:
The logic I used here is, "What parent libraries are directly asked for by libraries and binaries in this package? List the packages which have those libraries." 

I think the alternate logic you are suggesting is "What parent libraries are directly asked for by libraries and binaries in this package? List the packages which have those libraries, but don't list a package if another package listed would already provide that package."

The first algorithm guarantees packages won't break if the dependency tree implied in the second algorithm changes (through a package getting dropped from a build because of updates, etc.) I think it's a safe middle ground between "list all dependencies recursively as suggested by `ldd`" and "list the minimum possible set of dependencies."

Thoughts?

Reply:
I lean in the direction of the minimum possible set of dependencies.  If we test packages thoroughly after each update, we should be fine.

Reply:
But not putting it there doesn't make it a smaller list of dependencies. That's just _hiding a dependency_. Listing the actual dependencies also gives insight into what the package is doing and wants, which I also think is a useful element of looking at a package file list of dependencies. Knowing that this package is calling both libjpeg and libtiff is _useful._

Reply:
I think it's useful for package maintainers but we should be able to _see_ the child dependencies with some kind of script without the need to list verbatim, right?

Reply:
Are we not the package maintainers? Are we not the target audience for what's inside these files?

Reply:
I just don't want us to get carried away with the "unnecessary" dependencies because it becomes difficult to maintain when every package has redundancies especially when the dependencies keep changing.

Reply:
This is literally the reason I'm trying to keep all immediate/direc/parent dependencies in each package file.

> because it becomes difficult to maintain when every package has redundancies especially when the dependencies keep changing.

This is literally the reason I'm trying to keep all immediate/direc/parent dependencies in each package file. :)